airflow/providers/facebook/ads/hooks/ads.py

```diff
@@ -0,0 +1,146 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains Facebook Ads Reporting hooks
+"""
+import time
+from enum import Enum
+from typing import Any, Dict, List
+
+from cached_property import cached_property
+from facebook_business.adobjects.adaccount import AdAccount
+from facebook_business.adobjects.adreportrun import AdReportRun
+from facebook_business.adobjects.adsinsights import AdsInsights
+from facebook_business.api import FacebookAdsApi
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+class JobStatus(Enum):
+    """
+    Available options for facebook async task status
+    """
+    COMPLETED = 'Job Completed'
+    STARTED = 'Job Started'
+    RUNNING = 'Job Running'
+    FAILED = 'Job Failed'
+    SKIPPED = 'Job Skipped'
+
+
+class FacebookAdsReportingHook(BaseHook):
+    """
+    Hook for the Facebook Ads API
+
+    .. seealso::
+        For more information on the Facebook Ads API, take a look at the API docs:
+        https://developers.facebook.com/docs/marketing-apis/
+
+    :param facebook_conn_id: Airflow Facebook Ads connection ID
+    :type facebook_conn_id: str
+    :param api_version: The version of Facebook API. Default to v6.0
+    :type api_version: str
+
+    """
+
+    def __init__(
+        self,
+        facebook_conn_id: str = "facebook_default",
+        api_version: str = "v6.0",
+    ) -> None:
+        super().__init__()
+        self.facebook_conn_id = facebook_conn_id
+        self.api_version = api_version
+        self.client_required_fields = ["app_id",
+                                       "app_secret",
+                                       "access_token",
+                                       "account_id"]
+        self.default_params = {
+            'level': 'ad',
+            'date_preset': 'yesterday',
+        }
+
+    def _get_service(self) -> FacebookAdsApi:
+        """ Returns Facebook Ads Client using a service account"""
+        config = self.facebook_ads_config
+        missings = []
+        for _each in self.client_required_fields:
```

Comment:
This is very important, but can you inline this constant or make it a class attribute rather than an instance attribute? This is confusing,